### PR TITLE
Backport improvements made to Keychain in 10.0

### PR DESCRIPTION
**Pull request by andreparames**
After the migration there has been some work done on the Keychain module in branch 10.0, this PR ports it back to branch 9.0

### CODE REVIEW

File: keychain/tests/test_keychain.py [hunk 1]
```diff
@@ -21,6 +24,7 @@ def setUp(self):
 
         self.keychain = self.env['keychain.account']
         config['keychain_key'] = Fernet.generate_key()
+        config['running_env'] = ""
```
Comment (lmignon) [inline keychain/tests/test_keychain.py]: @andreparames IMO this commit must be removed... Indeed, you reset the running_env before before to save its previous value at the next line.

**Review by lmignon — CHANGES_REQUESTED** [review]

File: keychain/tests/test_keychain.py (hunk 1 above)
Comment (andreparames) [reply keychain/tests/test_keychain.py]: You're right, I took the "Mig V10" commit and removed the actual migration parts, without realizing what was left was no longer relevant. Will remove the commit.

**Review by lmignon — APPROVED** [review]
LGTM (Code review no test)

**Comment by lmignon** [comment]
@andreparames Thank you for this backport!

**Review by ThomasBinsfeld — APPROVED** [review]
Code review and functional tests :+1:

**Pull request merged**